tests/conftest.py

```diff
@@ -303,26 +303,41 @@ def _slurm_not_installed():
 def _subprocess_setup(request, cfg=None, uwsgi=False):
     """setup the supervisor"""
     import os
+    import sirepo.job
+    import sirepo.pkcli.service
     from pykern.pkcollections import PKDict
 
+    ports = PKDict()
+
+    # different port than default so can run tests when supervisor running
+    def _next_conftest_port(port_key, port_pkg):
+        p = ports.get(port_key)
+        if p:
+            ports[port_key] = p + 1
```

Comment:
This is a bit wonky. Let's define all cases in PORT_DEFAULTS so we don't have this type of thing and instead assert ports in all cases. 

I think the code in _job_supervisor_check could be generalized to test for the socket so we can ask the question here if the socket is in use. If it is in use, then we just increment by the default amount. This would simplify the problem of collisions. Perhaps that's the best approach. Thoughts?

Reply:
Would there be a loop to keep checking the next port (to some limit) or one and done?

Reply:
Correct. You would loop until you found a free port.

Reply:
You need to close the port once if is opened. You can continue to bind the same socket, btw, so the try/except only needs to be around the bind.